ably/rest_client.go

```diff
@@ -157,17 +157,18 @@ func (c *REST) Time() (time.Time, error) {
 // Stats gives the channel's metrics according to the given parameters.
 // The returned result can be inspected for the statistics via the Stats()
 // method.
-func (c *REST) Stats(params *PaginateParams) (*PaginatedResult, error) {
-	return newPaginatedResult(nil, paginatedRequest{typ: statType, path: "/stats", params: params, query: query(c.get), logger: c.logger(), respCheck: checkValidHTTPResponse})
+func (c *REST) Stats(ctx context.Context, params *PaginateParams) (*PaginatedResult, error) {
+	return newPaginatedResult(ctx, nil, paginatedRequest{typ: statType, path: "/stats", params: params, query: query(c.get), logger: c.logger(), respCheck: checkValidHTTPResponse})
 }
 
 // request this contains fields necessary to compose http request that will be
 // sent ably endpoints.
 type request struct {
-	Method string
-	Path   string
-	In     interface{} // value to be encoded and sent with request body
-	Out    interface{} // value to store decoded response body
+	Context context.Context
```

Comment:
This context should be passed via the call stack, not stored as a struct field

Reply:
Not sure why I did that, it wasn't even used. Fixed at c8a85d68fba44cd6f45c225702154cc15ecd0f51.